src/workflow/StreamlitUI.py

```
@@ -852,7 +854,58 @@ def input_TOPP(
+    ) -> None:
+        self._input_TOPP_impl(
+            topp_tool_name, num_cols, exclude_parameters, include_parameters,
+            flag_parameters, display_tool_name, display_subsections,
+            display_subsection_tabs, custom_defaults, tool_instance_name,
+        )
+
+    def _input_TOPP_impl(
+        self,
+        topp_tool_name: str,
+        num_cols: int = 4,
+        exclude_parameters: List[str] = [],
```

Ruff (0.15.21)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)